honeybadgermpc/hyper_invertible.py

```diff
@@ -0,0 +1,104 @@
+import asyncio
+import logging
+from honeybadgermpc.exceptions import HoneyBadgerMPCError
+from honeybadgermpc.polynomial import EvalPoint, polynomials_over
+from honeybadgermpc.reed_solomon import Algorithm, EncoderFactory, DecoderFactory
+from honeybadgermpc.batch_reconstruction import subscribe_recv, wrap_send
+
+
+class HyperInvMessageType(object):
+    SUCCESS = "S"
+    ABORT = "A"
+
+
+async def _recv_loop(n, recv, s=0):
+    results = [None]*n
+    for _ in range(n):
+        sender_id, value = await recv()
```

Comment:
could we do this with a gather?

Reply:
I could, but this will make the code a bit less readable since I will have to gather and then iterate again to assign the values correctly. This way it is also easier to debug since you can add a log to view what values are being received. I think it is better kept this way.